google/cloud/storage/internal/object_requests.h

```diff
@@ -212,6 +212,35 @@ class UpdateObjectRequest
 
 std::ostream& operator<<(std::ostream& os, UpdateObjectRequest const& r);
 
+/**
+ * Represents a request to the `Objects: compose` API.
+ */
+class ComposeObjectRequest
+    : public GenericObjectRequest<ComposeObjectRequest, EncryptionKey,
+                                  Generation, DestinationPredefinedAcl,
+                                  IfGenerationMatch, IfMetagenerationMatch,
+                                  SourceEncryptionKey, UserProject> {
+ public:
+  ComposeObjectRequest() = default;
+  explicit ComposeObjectRequest(
+      std::string bucket_name, std::string destination_object_name,
```

Comment:
Following the discussion on #1085, please reverse the order of the arguments so it is "source" arguments first, destination arguments next.